R/class-Collections.R

```diff
@@ -0,0 +1,415 @@
+#' object for use in a Microbiome Dataset. 
+#' @param name The name of the collection
+#' @param data A data.frame or a character vector representing a file path to a data.frame
+#' @param recordIdColumn The name of the column in the data.frame that contains the record id
+#' @param ancestorIdColumns A character vector of column names representing parent entities of the recordIdColumn
+#' @export
+#' @rdname Collection-methods
+setGeneric("Collection", function(name, data, recordIdColumn, ancestorIdColumns) standardGeneric("Collection"))
+
+#' @rdname Collection-methods
+#' @aliases Collection,character,data.frame,character,character-method
+setMethod("Collection", signature("character", "data.frame", "character", "character"), function(name, data, recordIdColumn, ancestorIdColumns) {
+    data <- data.table::setDT(data)
+    new("Collection", name = name, data = data, recordIdColumn = recordIdColumn, ancestorIdColumns = ancestorIdColumns)
+})
+
+#' @rdname Collection-methods
+#' @aliases Collection,character,data.frame,missing,missing-method
+setMethod("Collection", signature("character", "data.frame", "missing", "missing"), function(name, data, recordIdColumn, ancestorIdColumns) {
+    warning("Id columns not specified, assuming first column is record id and other Id columns end in `_Id`")
+    recordIdColumn <- findRecordIdColumn(names(data))
+    ancestorIdColumns <- findAncestorIdColumns(names(data))
+    data <- data.table::setDT(data)
+    new("Collection", name = name, data = data, recordIdColumn = recordIdColumn, ancestorIdColumns = ancestorIdColumns)
+})
+
+#' @rdname Collection-methods
+#' @aliases Collection,character,character,missing,missing-method
+setMethod("Collection", signature("character", "character", "missing", "missing"), function(name, data, recordIdColumn, ancestorIdColumns) {
+    data <- data.table::fread(data)
+    warning("Id columns not specified, assuming first column is record id and other Id columns end in `_Id`")
+    recordIdColumn <- findRecordIdColumn(names(data))
+    ancestorIdColumns <- findAncestorIdColumns(names(data))
+    new("Collection", name = name, data = data, recordIdColumn = recordIdColumn, ancestorIdColumns = ancestorIdColumns)
+})
+
+#' @rdname Collection-methods
+#' @aliases Collection,character,character,character,character-method
+setMethod("Collection", signature("character", "character", "character", "character"), function(name, data, recordIdColumn, ancestorIdColumns) {
+    data <- data.table::fread(data)
+    new("Collection", name =name, data = data, recordIdColumn = recordIdColumn, ancestorIdColumns = ancestorIdColumns)
+})
+
+#' @rdname Collection-methods
+#' @aliases Collection,missing,missing,missing,missing-method
+setMethod("Collection", signature("missing", "missing", "missing", "missing"), function(name, data, recordIdColumn, ancestorIdColumns) {
+    new("Collection")
+})
+
+
+##### some helpers for building Collections objects from various data sources #####
```

Comment:
Worth a new file for organizational purposes. A user looking at this wouldn't be totally sure if it's for them or for us. With a new file we can better explain that these are all for us helping to build Collections. Also anytime we talk about "downloads" we should specify from the mbio site etc. Often we have "eda services" explicitly stated which i think is really helpful so that outsiders browsing the code know what is/isn't something they should worry about

Reply:
internal-utils.R ?? really id say if its not exported, users shouldnt assume its for them. but i get thats maybe not enough for some people..

Reply:
lol i'll go ahead and say i'm not that savvy. I like internal-utils.R! That's obvious enough that even i wouldn't poke around in there